### PR TITLE
Docs: Update components documentation

### PR DESCRIPTION
**Pull request by stephanethomas**
This small pull request is a follow-up of p1521140291000512-calypso-design which updates the documentation about components in an attempt to clarify the different types used in Calypso.

### CODE REVIEW

**Comment by stephanethomas** [comment]
A heads-up that I've just force pushed a small change.

File: client/components/README.md [hunk 1]
```diff
@@ -1,8 +1,10 @@
 Components
 ==========
 
-_Read the [documentation on components](https://github.com/Automattic/wp-calypso/blob/master/docs/components.md) for more in-depth information._
+Components are React components used for composing the UI of Calypso.
```
Comment (gwwar) [inline client/components/README.md]: Should we add a few notes on how they behave more as UI primitives?

For example we'd usually be able to write a component as a functional stateless component (provided that folks didn't need to keep track of refs).

File: client/components/README.md [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
 
-This place harbors shared React components used for composing the UI of Calypso. Components come with their own styles defined [according to our guidelines](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/css.md), and manually loaded from the [styles assets folder](https://github.com/Automattic/wp-calypso/blob/master/assets/stylesheets/_components.scss). Structuring the user interface with these building blocks has several benefits — like allowing to quickly construct a view that is visually consistent with the rest of Calypso, and easier to iterate on.
+They come with their own styles defined [according to our guidelines](../../docs/coding-guidelines/css.md), and manually loaded from the [styles assets folder](../../assets/stylesheets/_components.scss). Structuring the user interface with these building blocks has several benefits — like allowing to quickly construct a view that is visually consistent with the rest of Calypso, and easier to iterate on.
```
Comment (gwwar) [inline client/components/README.md]: Maybe choose a different word than `block` in `building blocks`?

File: docs/components.md [hunk 3]
```diff
@@ -6,7 +6,15 @@ Our use of composition is one that embraces CSS reusability, with the cascade as
 
 ## Glossary
 
-This document focuses on UI components, not data containers or other non-UI pieces. That is, React components that directly render HTML elements and require the addition of class attributes for styling. We have essentially two kinds of these:
+You will encounter the following types of components in Calypso:
+
+* [UI components](../client/components/README.md) (UI primitives)
+* [Blocks](../client/blocks/README.md) (components which are connected to state, or otherwise directly represent application entities)
```
Comment (gwwar) [inline docs/components.md]: Good call!

File: client/components/README.md (hunk 2 above)
Comment (stephanethomas) [reply client/components/README.md]: Haha. I had the same thought when refining this section, but couldn't come up with a better term.

**Comment by gwwar** [comment]
Don't forget to merge @stephanethomas !

**Pull request merged**